.circleci/config.yml

```diff
@@ -99,11 +99,32 @@ jobs:
           command: |
             mkdir -p /tmp/test-results
             trap "go-junit-report < /tmp/test-results/go-test.out > /tmp/test-results/go-test-report.xml" EXIT
-            go run build/ci.go test -v | tee /tmp/test-results/go-test.out
+            go run build/ci.go test -v -race | tee /tmp/test-results/go-test.out
       - store_artifacts:
           path: /tmp/test-results
           destination: raw-test-output
+      - store_test_results:
+          path: /tmp/test-results
 
+  unit-tests-race:
+    executor: golang
+    resource_class: medium+
+    steps:
+      - attach_workspace:
+          at: ~/repos
+      - restore_cache:
+          keys:
+            - go-mod-v1-{{ checksum "go.sum" }}
+      - run: go get github.com/jstemmer/go-junit-report
+      - run:
+          name: Run Tests
+          command: |
+            mkdir -p /tmp/test-results
+            trap "go-junit-report < /tmp/test-results/go-test.out > /tmp/test-results/go-test-report.xml" EXIT
+            go run build/ci.go test -v | tee /tmp/test-results/go-test.out
```

Comment:
I'm actually thinking maybe its clearer to just run go test directly, build/ci.go obfuscates the actual test command that is being run and as far as I can see doesn't add anything. If we did that we wouldn't have to change build/ci.go which would help prevent future merge conflicts.

Reply:
I think it makes more sense to me to keep using `build/ci.go` as a proxy. As for upstream conflicts, I don't think it would create tricky situation here.

Reply:
Why would you prefer to use `build/ci.go` over `go test`?

Reply:
Not a preference, just intended to keep it consistent. 
I think here `build/ci.go` acts as an interface that is handling all build related scripts under the hood, even though some of its function is just an "air wrapper" that does no additional work. But I value this design philosophy.
If we want to expose details like `go test` rather than using this interface, then it would make sense to replace all the `build/ci.go` scripts with actual scripts, and that doesn't seem right.

Reply:
Hey @YoduYodu so we talked about this and IIRC we decided to switch all test calls to use go test rather than ci.go. Was this something you were considering doing as part of this PR or separately?

Reply:
Yes I will do this as part of this PR